sermant-agentcore/sermant-agentcore-implement/src/test/java/io/sermant/implement/service/xds/handler/EdsXdsTest.java

```diff
@@ -0,0 +1,144 @@
+package io.sermant.implement.service.xds.handler;
+
+import com.google.protobuf.Any;
+
+import io.envoyproxy.envoy.config.core.v3.Address;
+import io.envoyproxy.envoy.config.core.v3.SocketAddress;
+import io.envoyproxy.envoy.config.endpoint.v3.ClusterLoadAssignment;
+import io.envoyproxy.envoy.config.endpoint.v3.Endpoint;
+import io.envoyproxy.envoy.config.endpoint.v3.LbEndpoint;
+import io.envoyproxy.envoy.config.endpoint.v3.LocalityLbEndpoints;
+import io.envoyproxy.envoy.service.discovery.v3.DiscoveryResponse;
+import io.sermant.core.service.xds.entity.ServiceInstance;
+import io.sermant.core.service.xds.listener.XdsServiceDiscoveryListener;
+import io.sermant.implement.service.xds.BaseXdsTest;
+import io.sermant.implement.service.xds.cache.XdsDataCache;
+import io.sermant.implement.service.xds.env.XdsConstant;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * EdsHandler UT
+ *
+ * @author daizhenyu
+ * @since 2024-05-25
+ **/
+public class EdsXdsTest extends BaseXdsTest {
+    private static String serviceName = "serviceA";
+
+    private String clusterName = "outbound|8080||serviceA.default.svc.cluster.local";
+
+    private static EdsHandler handler;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        handler = new EdsHandler(client, XdsConstant.EDS_RESOURCE_TYPE);
+        Mockito.doReturn(requestStreamObserver).when(client).getDiscoveryRequestObserver(handler.getResponseStreamObserver(serviceName, null));
+
+        handler.subscribe(serviceName, null);
+        XdsDataCache.SERVICE_DISCOVER_LISTENER.put(serviceName, Arrays.asList(new XdsServiceDiscoveryListenerImpl()));
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        Mockito.clearAllCaches();
+        XdsDataCache.REQUEST_OBSERVERS.clear();
+    }
+
+    @Test
+    public void testHandleResponse() {
+        // first update service instance of serviceA
+        handler.handleResponse(serviceName, buildDiscoveryResponse("127.0.0.1", 8080));
+        Set<ServiceInstance> serviceInstances = XdsDataCache.SERVICE_INSTANCES.get(serviceName);
+        List<XdsServiceDiscoveryListener> listeners = XdsDataCache.SERVICE_DISCOVER_LISTENER
+                .get(serviceName);
+        XdsServiceDiscoveryListenerImpl listenerImpl =
+                (XdsServiceDiscoveryListenerImpl) listeners.get(0);
+
+        Assert.assertEquals(1, serviceInstances.size());
+        Assert.assertEquals(1, listenerImpl.getCount());
+        Iterator<ServiceInstance> iterator = serviceInstances.iterator();
+        if (iterator.hasNext()) {
+            ServiceInstance next = iterator.next();
+            Assert.assertEquals("127.0.0.1", next.getHost());
+            Assert.assertEquals(8080, next.getPort());
+        }
```

Comment:
The two verification logics are the same. The commission method is recommended.

Reply:
done